packages/itwinui-css/src/tabs/base.scss

```diff
@@ -186,37 +184,38 @@ $borderless-horizontal-tab-min-height: calc(
   block-size: 100%;
 
   .iui-tabs {
-    li,
     .iui-tab {
       inline-size: 100%;
       white-space: nowrap;
+      block-size: auto;
     }
 
     ~ .iui-tabs-content {
       flex-grow: 1;
       overflow: auto;
     }
 
-    &[data-iui-overflow='true'] {
-      overflow-y: auto;
-      min-inline-size: min-content;
+    overflow-y: auto;
+    animation-name: scroll-shadow-inset-vertical;
+    animation-timing-function: linear;
+    /* stylelint-disable-next-line property-no-unknown */
+    animation-timeline: scroll(self inline);
 
-      &[data-iui-scroll-placement='start'] {
-        /* stylelint-disable-next-line property-no-vendor-prefix */
-        -webkit-mask-image: linear-gradient(0deg, transparent 0%, rgb(0, 0, 0) 5%, rgb(0, 0, 0) 100%, transparent 100%);
-        mask-image: linear-gradient(0deg, transparent 0%, rgb(0, 0, 0) 5%, rgb(0, 0, 0) 100%, transparent 100%);
+    @keyframes scroll-shadow-inset-vertical {
```

Comment:
non-blocking: I am slightly confused about how the two animations work. Could you elaborate?